### DISCUSSION TITLE
Ongoing Explorer-crash issue related somehow to permissions, it seems.

### DISCUSSION THREAD

**Discussion by CLBrown66** [Open-Shell/Open-Shell-Menu#1853] [General]
Ongoing Explorer-crash issue (related somehow to permissions, it seems.)

Let me preface this by saying that I'm NOT an "expert programmer" with the current generation of Windows-based software tools, nor of the most intimate details of how modern Windows operates in general.  I'm far, far more knowledgable than the typical user, certainly, but the "guts" of how Windows does things often escapes me.  This is one such example.  I know JUST ENOUGH to figure out what the problem seems to be, but not enough to have any idea how to resolve it.

I have the most recent "stable" release of OpenShell installed on my PC.  I was running Win7 up until VERY recently, but had an operating system "barf" which somehow resulted in my no longer having sound (despite devices being installed and working in "device manager" and being able to do their own internal diagnostics, Windows Audio always reported "no devices installed.")  My solution was to finally bite the bullet and do a clean-sheet install of Windows 10 (my machine won't support Win11, at least not officially).

It hasn't gone super-well.  I'm actually starting to consider reinstalling Win7.  It's so much easier to work with.  And the Microsoft obsession with blocking end-users from having control over their own machines seems to be at the heart of all of the issues I'm encountering.

That seems, as well, to be the issue here.  With OpenShell installed, initially, everything seemed fine.  But I have a LOT of programs on my PC (always have, always will... it's MY computer, not Microsoft's, and I can install as much on here as I like, after all).  So far, I've only installed "officially Windows 10 compliant" stuff, though I do have a lot of older software I'd also like to get reinstalled.  I have multiple hard disks on the PC, with the first one being "Main" (OS and important or frequently used applications), "Work" (where I keep my CAD programs, rendering programs, and various productivity stuff), "Games" (where I keep locally installed disk-based or downloaded game software), "Online Games" (with Steam, GOG, Epic, Ubisoft, etc, online game platforms and their associated game directories), plus "Media" (for all my photos, videos, music, etc) and "Software" (where I keep the majority of my downloaded software installation files and related materials).  I also back all the media and software stuff up to a Synology NAS.

So... I have a pretty huge set of files and applications on this PC, spread over multiple disks.  And a commensurately large "Start Menu."  That menu structure is multi-level, tiered, and very well organized.  For example, if I want to run "Falcon 4.0," I go to "Start Menu - Programs - Recreation - Sims - Flight - Falcon Series - Falcon 4.0.  Within that folder are three subdirectories - "Docs & Links," "Media," and "Utilities" apart from the main program itself at the root level of that directory.  

This worked marvelously as a system waaay back in Win9x, and with WinXP.  It also worked just fine in Win7, under "all programs."  I preferred the Classic Shell/Open Shell ability to use the simplified "classic start menu" format, but could work just fine with the "all programs" under the original Win7 menu as well.

Not so with Win10/Win11, however, which have removed all capabilities of "nested folders" within the start menu.  So, a system like "Open Shell" is absolutely MANDATORY in order for me to be able to use my system as I want to (actually, "need to.")

All that's just background to understand what I'm trying to do.  Now, to the issues.

The program itself SEEMS to work just fine, except that I get an "explorer crash and restart" every few minutes.  This is a HUGE annoyance.  If I'm trying to do something immersive, I need to shut down OpenShell Start Menu to avoid interrupting.  And even while working "in a window" or the like, I periodically have to stop while Explorer resets itself.  Of course, my "tray icons" don't all come back... only about half do... with the only way to get them all back seemingly to log out and log back in.

NOT a tenable situation.

So, going to the Computer Management - Event Viewer - Windows Logs stuff to see what's going on, I see a LOT of errors.

Under "System" I get this, which I strongly suspect is the key of the issue:

***************************************************************************

Log Name:  System
Source:  DistributeCOM
Event ID:  10016
Level:  Warning
User:  LOCAL SERVICE
OpCode:  Info
Logged:  2/21/2024 7:19:05 AM
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

****************************************
The application-specific permission settings do not grant Local Activation permission for the COM Server application with CLSID 
{6B3B8D23-FA8D-40B9-8DBD-B950333E2C52}
 and APPID 
{4839DDB7-58C2-48F5-8283-E1D1807D0D7D}
 to the user NT AUTHORITY\LOCAL SERVICE SID (S-1-5-19) from address LocalHost (Using LRPC) running in the application container Unavailable SID (Unavailable). This security permission can be modified using the Component Services administrative tool.
***************************************************************************
Checking through my registry, I find that the CLSID and APPID above are related to something called "Immersive Shell."

I see this error regularly... not just a single instance, but quite a few, and seemingly related to the crash I observe by time of event.

And in my "Application" event log, I see many instances of THIS, seemingly at the same time code (though this one is from a different event.  (I often see multiple of these related to a single instance of what I list above, by the way, at the same approximate time code.)  The text tells me that these are likely related, however.

***************************************************************************
Log Name:  Application
Source:  SideBySide
Event ID:  33
Level:  Error
User:  N/A
OpCode:  Info
Logged:  2/21/2024 8:52:22 AM
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

******************************************

Activation context generation failed for "F:\Steam\SteamApps\common\america's army 3\Binaries\AA3Loader.exe". Dependent Assembly Microsoft.VC80.MFC,processorArchitecture="x86",publicKeyToken="1fc8b3b9a1e18e3b",type="win32",version="8.0.50727.6195" could not be found. Please use sxstrace.exe for detailed diagnosis.

***************************************************************************

I also see quite a few which read like this:

***************************************************************************

Log Name:  Application
Source:  SideBySide
Event ID:  78
Level:  Error
User:  N/A
OpCode:  Info
Logged:  2/21/2024 8:52:40 AM
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

*********************************
Activation context generation failed for "E:\1stPersn\Quake Series\Enemy Territory - QUAKE Wars\serverlauncher.exe".Error in manifest or policy file "" on line . A component version required by the application conflicts with another component version already active. Conflicting components are:. Component 1: C:\WINDOWS\WinSxS\manifests\x86_microsoft.windows.common-controls_6595b64144ccf1df_6.0.19041.3636_none_a863d714867441db.manifest. Component 2: C:\WINDOWS\WinSxS\manifests\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.19041.3636_none_60b6a03d71f818d5.manifest.

***************************************************************************

These two errors then inevitably lead to the following trio of "information" messages

***************************************************************************

Log Name:  Application
Source:  Windows Error Reporting
Event ID:  1001
Level:  Information
User:  N/A
OpeCode:  Info
Logged:  2/21/2024 8:52:08
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

**************************************

Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: explorer.exe
P2: 10.0.19041.4046
P3: 7ede0e20
P4: StartMenuDLL.dll
P5: 4.4.191.0
P6: 64dcd337
P7: c0000005
P8: 00000000000fe3a6
P9: 
P10: 

Attached files:

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_explorer.exe_bb15bbeaf889a2b1682c3a5cab87ab64273a4e_4a37867f_8bc6a3ce-317b-41b7-9d36-4179693f8874

Analysis symbol: 
Rechecking for solution: 0
Report Id: 31d540f9-9d1e-47de-bd49-530ab98f29d8
Report Status: 4
Hashed bucket: 
Cab Guid: 0

***************************************************************************

Log Name:  Application
Source:  Windows Error Reporting
Event ID:  1001
Level:  Information
User:  N/A
OpCode:  Info
Logged:  2/21/2024 8:52:08 AM
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

**************************************

Fault bucket 2230297917718733442, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: explorer.exe
P2: 10.0.19041.4046
P3: 7ede0e20
P4: StartMenuDLL.dll
P5: 4.4.191.0
P6: 64dcd337
P7: c0000005
P8: 00000000000fe3a6
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA6D3.tmp.WERInternalMetadata.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_explorer.exe_bb15bbeaf889a2b1682c3a5cab87ab64273a4e_4a37867f_8bc6a3ce-317b-41b7-9d36-4179693f8874

Analysis symbol: 
Rechecking for solution: 0
Report Id: 31d540f9-9d1e-47de-bd49-530ab98f29d8
Report Status: 268435456
Hashed bucket: 133d21cab8f7c6da2ef39c23b8eb2282
Cab Guid: 0

***************************************************************************

Log Name:  Application
Source:  Winlogon
Event ID:  1002
Level:  Information
User:  N/A
OpCode:  Info
Logged:  2/21/2024 8:52:09 AM
Task Category:  None
Keywords:  Classic
Computer:  CarysPCWin10

**************************************

The shell stopped unexpectedly and explorer.exe was restarted.

***************************************************************************

Now, this cycles through, over and over, repeatedly, unless I "kill" OpenShell's start menu (which I do via Process Explorer) or unless it simply shuts down on its own (which happens fairly frequently as a result of these cycling crashes).

This is a HUGE issue, and I can't seem to figure out how to get around it.  There are intricacies of how Windows does permissions and "security" which I simply don't understand, much less do I have any idea how to resolve them.

But it's pretty clear that somehow OpenShell's "StartMenuDLL.dll" and "Explorer.exe" are not playing nicely together.  Every error I see has that "Keywords:  Classic" present, which I am interpreting to mean that this is specific to "classic shell" (or rather, now, "Open Shell.")  And it seems that it's all related somehow to "activation permissions."

I'm stumped.  But if I'm going to continue to run on Windows 10, rather than "downgrading" back to Windows 7... I really need to figure out how to resolve this.  It really doees seem to be a matter of somehow getting "DCom" "activation permissions" assigned t "StartMenuDLL.dll."  But I have NO FREAKING CLUE how to go about doing that, or even what that means.

I really hope someone here can help me out.

**Comment by ge0rdi** [comment]
Please, provide crash dump obtained using this [guide](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-create-dump-on-Windows-Explorer-crash).

**Comment by CLBrown66** [comment]
Sorry, I responded to you directly, as the response came to me as an email....   it's hard to work effectively when explorer crashes every few seconds.  I've had to shut down OpenShell in order to get this response started.

I've tried what was suggested in the guide, above, and it hasn't worked, for some reason.

Here, I'm following instructions related to starting a procdump.

![hfijafjgilhgmfgd](https://github.com/Open-Shell/Open-Shell-Menu/assets/69014922/e446c33d-da67-4195-bad5-ffaaf83004dc)

And here's the status of my crashdumps folder (where the dumps ought to be showing up, right?) after restarting OpenShell and letting it cycle through multiple crash cycles.

![gpbolnbhdanjoclb](https://github.com/Open-Shell/Open-Shell-Menu/assets/69014922/5645bfd8-ee80-4fac-964c-8eadb346a611)

So, "no joy" so far.  If I can't even get the diagnostic tools to work...  and I presume that "multiple processes with the same name" thing is why it's not working???...  I don't know what else to try.

**Reply by ge0rdi** [reply]
Could you try to register _local dumps_ as described in the second part of guide?
> In case of random crashes that are hard to replicate, you can instruct Windows to create dump into specified location using [Collecting User-Mode Dumps](https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps) guide
